### PR TITLE
[tests-only][full-ci] add tests for new role space editor without versions

### PR DESCRIPTION
**Pull request by prashant-gurung899**
## Description
This PR adds test for new permission role `Space Editor Without Versions`
``` feature
Scenario: check the versions of a file in a shared space as editor without versions role
```

## Related Issue

- Related to: https://github.com/owncloud/ocis/issues/9955

## How Has This Been Tested?

- locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/TestHelpers/GraphHelper.php [hunk 1]
```diff
@@ -16,6 +16,21 @@
  * A helper class for managing Graph API requests
  */
 class GraphHelper {
+	public static array $DEFAULT_PERMISSIONS_ROLES = [
```
Comment (saw-jan) [inline tests/acceptance/TestHelpers/GraphHelper.php:19]: let's make it a constant varaible
```suggestion
	const DEFAULT_PERMISSIONS_ROLES = [
```

File: tests/acceptance/TestHelpers/GraphHelper.php [hunk 2]
```diff
@@ -16,6 +16,21 @@
...
+		'Viewer' => 'b1e2218d-eef8-4d4c-b82d-0f1a1b48f3b5',
+		'Editor' => 'fb6c3e19-e378-47e5-b277-9732f9de6e21',
+		'File Editor' => '2d00ce52-1fc2-4dbc-8b95-a73b73395f5a',
+		'Uploader' => '1c996275-f1c9-4e71-abdf-a42f6495e960',
+		'Space Viewer' => 'a8d5fe5e-96e3-418d-825b-534dbdf22b99',
+		'Space Editor' => '58c63c02-1d89-4572-916a-870abc5a1b7d',
+		'Manager' => '312c0871-5ef7-4b3a-85b6-0e4074c64049',
+	];
+
+	public static array $ADDITIONAL_PERMISSIONS_ROLES = [
```
Comment (saw-jan) [inline tests/acceptance/TestHelpers/GraphHelper.php:29]:
```suggestion
	const ADDITIONAL_PERMISSIONS_ROLES = [
```

File: tests/acceptance/TestHelpers/GraphHelper.php [hunk 3]
```diff
@@ -1657,31 +1672,13 @@ public static function getPermissionsList(
...
-			case 'Uploader':
-				return '1c996275-f1c9-4e71-abdf-a42f6495e960';
-			case 'Manager':
-				return '312c0871-5ef7-4b3a-85b6-0e4074c64049';
-			case 'Secure viewer':
-				return 'aa97fe03-7980-45ac-9e50-b325749fd7e6';
-			default:
-				throw new \Exception('Role ' . $permissionsRole . ' not found');
+	public static function getPermissionsRoleIdByName(string $permissionsRole): string {
+		$all_available_roles = array_merge(self::$DEFAULT_PERMISSIONS_ROLES, self::$ADDITIONAL_PERMISSIONS_ROLES);
```
Comment (saw-jan) [inline tests/acceptance/TestHelpers/GraphHelper.php:1676]: we do not prefer snake_case for local varaibles.
```suggestion
		$allPermissionsRoles = array_merge(self::$DEFAULT_PERMISSIONS_ROLES, self::$ADDITIONAL_PERMISSIONS_ROLES);
```

File: tests/acceptance/TestHelpers/GraphHelper.php [hunk 4]
```diff
@@ -1693,29 +1690,12 @@ public static function getPermissionsRoleIdByName(
...
-				return 'Space Editor';
-			case '312c0871-5ef7-4b3a-85b6-0e4074c64049':
-				return 'Manager';
-			case 'aa97fe03-7980-45ac-9e50-b325749fd7e6':
-				return 'Secure viewer';
-			default:
-				throw new \Exception('Permission role id: ' . $permissionsRoleId . ' not found');
-		}
+	public static function getPermissionNameByPermissionRoleId(string $permissionsRoleId): string {
+		$all_available_roles = array_merge(self::$DEFAULT_PERMISSIONS_ROLES, self::$ADDITIONAL_PERMISSIONS_ROLES);
```
Comment (saw-jan) [inline tests/acceptance/TestHelpers/GraphHelper.php:1694]:
```suggestion
		$allPermissionsRoles = array_merge(self::$DEFAULT_PERMISSIONS_ROLES, self::$ADDITIONAL_PERMISSIONS_ROLES);
```

File: tests/acceptance/TestHelpers/GraphHelper.php [hunk 5]
```diff
@@ -1657,31 +1672,13 @@ public static function getPermissionsList(
...
-				return 'aa97fe03-7980-45ac-9e50-b325749fd7e6';
-			default:
-				throw new \Exception('Role ' . $permissionsRole . ' not found');
+	public static function getPermissionsRoleIdByName(string $permissionsRole): string {
+		$all_available_roles = array_merge(self::$DEFAULT_PERMISSIONS_ROLES, self::$ADDITIONAL_PERMISSIONS_ROLES);
+		if (\array_key_exists($permissionsRole, $all_available_roles)) {
+			return $all_available_roles[$permissionsRole];
 		}
+
+		throw new \Exception('Role ' . $permissionsRole . ' not found');
```
Comment (saw-jan) [inline tests/acceptance/TestHelpers/GraphHelper.php:1681]:
```suggestion
		throw new \Exception("Role '$permissionsRole' not found");
```

File: tests/acceptance/TestHelpers/GraphHelper.php [hunk 6]
```diff
@@ -1693,29 +1690,12 @@ public static function getPermissionsRoleIdByName(
...
-				return 'Secure viewer';
-			default:
-				throw new \Exception('Permission role id: ' . $permissionsRoleId . ' not found');
-		}
+	public static function getPermissionNameByPermissionRoleId(string $permissionsRoleId): string {
+		$all_available_roles = array_merge(self::$DEFAULT_PERMISSIONS_ROLES, self::$ADDITIONAL_PERMISSIONS_ROLES);
+		$roleName = array_search($permissionsRoleId, $all_available_roles);
+		return $roleName;
+
+		throw new \Exception('Permission role id: ' . $permissionsRoleId . ' not found');
```
Comment (saw-jan) [inline tests/acceptance/TestHelpers/GraphHelper.php:1698]:
```suggestion
		throw new \Exception("Permission role id: '$permissionsRoleId' not found");
```

Comment (saw-jan) [inline tests/acceptance/TestHelpers/GraphHelper.php:1698]: the throw statement will never reach.
```suggestion
		$roleName = array_search($permissionsRoleId, $all_available_roles);
		if ($roleName) {
			return $roleName;
		}

		throw new \Exception('Permission role id: ' . $permissionsRoleId . ' not found');
```

File: tests/acceptance/TestHelpers/GraphHelper.php [hunk 7]
```diff
@@ -2168,7 +2148,6 @@ public static function sendSharingInvitationForDrive(
 	): ResponseInterface {
 		$url = self::getBetaFullUrl($baseUrl, "drives/$spaceId/root/invite");
 		$body = self::createShareInviteBody($shareeIds, $shareTypes, $permissionsRole, $permissionsAction, $expirationDateTime);
-
```
Comment (saw-jan) [inline tests/acceptance/TestHelpers/GraphHelper.php:2171]: can be :arrows_counterclockwise:

File: tests/acceptance/bootstrap/OcisConfigContext.php [hunk 8]
```diff
@@ -75,6 +76,31 @@ public function theConfigHasBeenSetTo(string $configVariable, string $configValu
...
+	/**
+	 * @Given the role :role has been enabled
+	 *
+	 * @param string $role
+	 *
+	 * @return void
+	 */
+	public function theRoleHasBeenEnabled(string $role): void {
+		$roleId = GraphHelper::getPermissionsRoleIdByName($role);
+		$default_roleIDs = array_values(GraphHelper::$DEFAULT_PERMISSIONS_ROLES);
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/OcisConfigContext.php:88]:
```suggestion
		$defaultRoles = array_values(GraphHelper::$DEFAULT_PERMISSIONS_ROLES);
```

File: tests/acceptance/bootstrap/OcisConfigContext.php [hunk 9]
```diff
@@ -75,6 +76,31 @@ public function theConfigHasBeenSetTo(string $configVariable, string $configValu
...
+			$envs = [
+				"GRAPH_AVAILABLE_ROLES" => implode(',', $default_roleIDs),
+			];
+			$response =  OcisConfigHelper::reConfigureOcis($envs);
+			Assert::assertEquals(
+				200,
+				$response->getStatusCode(),
+				"Failed to enable role $role"
+			);
+		}
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/OcisConfigContext.php:101]:
```suggestion
		if (!\in_array($role, $default_roleIDs)) {
			$default_roleIDs[] = $roleId;
		}
		$envs = [
			"GRAPH_AVAILABLE_ROLES" => implode(',', $default_roleIDs),
		];
		$response =  OcisConfigHelper::reConfigureOcis($envs);
		Assert::assertEquals(
			200,
			$response->getStatusCode(),
			"Failed to enable role $role"
		);
```

File: tests/acceptance/features/apiSharingNgShareInvitation/shareInvitations.feature [hunk 10]
```diff
@@ -2248,7 +2248,6 @@ Feature: Send a sharing invitations
       | Space Viewer     | role not applicable to this resource                                                               |
       | Space Editor     | role not applicable to this resource                                                               |
       | Manager          | role not applicable to this resource                                                               |
-      | Co Owner         | Key: 'DriveItemInvite.Roles' Error:Field validation for 'Roles' failed on the 'available_role' tag |
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNgShareInvitation/shareInvitations.feature:2251]: this role doesn't exist. :+1:

File: tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature [hunk 11]
```diff
@@ -164,3 +164,14 @@ Feature: checking file versions using file id
       | source  | destination |
       | /       | folder/     |
       | folder/ | /           |
+
+  @env-config
+  Scenario: check the versions of a file in a shared space as editor without versions role
+    Given the role "Space Editor Without Versions" has been enabled
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpacesDavOperation/fileVersionByFileID.feature:170]:
```suggestion
    Given the administrator has enabled the permissions role "Space Editor Without Versions"
```

File: tests/acceptance/TestHelpers/GraphHelper.php (hunk 3 above)
Comment (prashant-gurung899) [reply tests/acceptance/TestHelpers/GraphHelper.php:1676]: Thanks for the info.

File: tests/acceptance/bootstrap/OcisConfigContext.php [hunk 12]
```diff
@@ -75,6 +76,31 @@ public function theConfigHasBeenSetTo(string $configVariable, string $configValu
...
 	}
 
+	/**
+	 * @Given the administrator has enabled the permissions role :role
+	 *
+	 * @param string $role
+	 *
+	 * @return void
+	 */
+	public function theRoleHasBeenEnabled(string $role): void {
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/OcisConfigContext.php:86]: method name should align with the step name
```suggestion
	public function theRoleHasBeenEnabled(string $role): void {
```

Comment (prashant-gurung899) [reply tests/acceptance/bootstrap/OcisConfigContext.php]: Thanks for the suggestion.

**Review by amrita-shrestha — APPROVED** [review]

**Pull request merged**